feature-toggles/feature-toggles-impl/src/test/java/com/duckduckgo/feature/toggles/codegen/ContributesRemoteFeatureCodeGeneratorTest.kt

```diff
@@ -115,6 +124,227 @@ class ContributesRemoteFeatureCodeGeneratorTest {
+                        "missingFeature": {
+                            "state": "enabled"
+                        },
+                        "fooFeature": {
+                            "state": "disabled"
+                        }
+                    }
+                }
+                """.trimIndent(),
+            ),
+        )
+        assertTrue(testFeature.self().isEnabled())
+        assertFalse(testFeature.fooFeature().isEnabled())
+    }
+
+    @Test
+    fun `re-evaluate feature when already preset in remote config but just added to client`() {
+        fun createAnotherFooFeature(): Any {
+            return Class
+                .forName("com.duckduckgo.feature.toggles.codegen.AnotherTestTriggerFeature_RemoteFeature")
+                .getConstructor(
+                    FeatureExceptions.Store::class.java,
+                    FeatureSettings.Store::class.java,
+                    dagger.Lazy::class.java as Class<*>,
+                    AppBuildConfig::class.java,
+                    VariantManager::class.java,
+                    Context::class.java,
+                ).newInstance(
+                    FeatureExceptions.EMPTY_STORE,
+                    FeatureSettings.EMPTY_STORE,
+                    Lazy { anotherTestFeature },
+                    appBuildConfig,
+                    variantManager,
+                    context,
+                )
+        }
+
+        assertFalse(anotherTestFeature.newFooFeature().isEnabled())
+
+        assertTrue(
+            (generatedFeatureNewInstance() as PrivacyFeaturePlugin).store(
+                "testFeature",
+                """
+                {
+                    "hash": "1",
+                    "state": "disabled",
+                    "features": {
+                        "newFooFeature": {
+                            "state": "enabled"
+                        },
+                        "fooFeature": {
+                            "state": "enabled"
+                        }
+                    }
+                }
+                """.trimIndent(),
+            ),
+        )
+        assertFalse(testFeature.self().isEnabled())
+        assertTrue(testFeature.fooFeature().isEnabled())
+        assertFalse(anotherTestFeature.newFooFeature().isEnabled())
+
+        assertTrue(
+            (createAnotherFooFeature() as PrivacyFeaturePlugin).store(
+                "testFeature",
+                """
+                {
+                    "hash": "1",
+                    "state": "disabled",
+                    "features": {
+                        "newFooFeature": {
+                            "state": "enabled"
+                        },
+                        "fooFeature": {
+                            "state": "enabled"
+                        }
+                    }
+                }
+                """.trimIndent(),
+            ),
+        )
+
+        assertTrue(anotherTestFeature.newFooFeature().isEnabled())
```

Comment:
Probably irrelevant in the context of this specific change, but isn't `anotherTestFeature.newFooFeature().isEnabled()` supposed to be false when `testFeature.self().isEnabled()` is false, regardless of the actual remote config value?

Reply:
The FF framework doesn't condition sub-feature `state` values to the root feature `state` value. This is left to the developer to decide if both `state` values should be affect each other or not